How to create an Apple PassType ID (#5)
abunsen/pn-mono: app/views/tutorial-files/apple-platform/how-to-create-an-apple-pass-type-id.md

title: How to Create an Apple Pass Type
author: @grundy

## Overview

> This guide is designed to help developers quickly generate pass types on Apple's Developer portal. A pass type consititutes a unique identifier and a signing certificate. Whether you're a seasoned developer or a nooB, this guide will get you through the steps like a ninja. We start with creating a pass type identifier for your organization and group of passes you want to distribute. This identifier, in the form of a reverse DNS string, is created in the Identifiers section of your Apple Developer portal. Once you have the pass type id, you'll need a signing certificate to go along with it. You get the signing certificate by creating a certificate signing request (CSR) on your PC and then generating the certificate in the Certificates section of the same portal. Once generated, you then can download the signing certificate to your machine for signing a pass bundle.

Reviewer: ```suggestion
This guide is designed to help developers quickly generate pass types on Apple's Developer portal. A pass type consititutes a unique identifier and a signing certificate. Whether you're a seasoned developer or a beginner, this guide will get you through the steps like a ninja. 

We start with creating a pass type identifier for your organization and group of passes you want to distribute. This identifier, in the form of a reverse DNS string, is created in the Identifiers section of your Apple Developer portal. Once you have the pass type id, you'll need a signing certificate to go along with it. You get the signing certificate by creating a certificate signing request (CSR) on your PC and then generating the certificate in the Certificates section of the Apple developer portal. 

Once generated, you then can download the signing certificate to your machine for signing a pass bundle.
```

## Create Pass Type ID

To start you'll need to create a pass type identifier in your Apple Developer account. This is done by filling out a simple form under the Identifiers section. The identifier should be a unique string, without spaces, typically following a reverse DNS pattern so that sorting a list of many identifiers groups them by organization instead of application.

> ### Go to [Apple Developer portal](https://developer.apple.com/)

Reviewer: ```suggestion
**Go to [Apple Developer portal](https://developer.apple.com/)**
```

![step1](tutorials/how-to-create-a-apple-pass-type-id/step1.jpg)

> ### Sign in to your developer account

Reviewer: ```suggestion
**Sign in to your developer account**
```

![step2](tutorials/how-to-create-a-apple-pass-type-id/step2.jpg)

> ### Select Identifiers and click Add (+)

Reviewer: ```suggestion
**Select Identifiers and click Add (+)**
```

![step3](tutorials/how-to-create-a-apple-pass-type-id/step3.jpg)

### Create a Pass Type Id

![step4](tutorials/how-to-create-a-apple-pass-type-id/step4.jpg)

> ### Register your Identifier with Description

Reviewer: ```suggestion
**Register your Identifier with description**
```

![step5](tutorials/how-to-create-a-apple-pass-type-id/step5.jpg)

## Generate Signing Certificate

Once a pass type identifier is registered on the Apple Developer portal, you'll need a certificate to start using it in making Apple Wallet passes. Similar to how application builds (.ipa files) need to be cryptographically signed by their developer before they can be installed on devices, Apple pass bundles (.pkpass files) need to be signed by their issuer before they can be installed in Apple Wallets. So you will need a signing certificate for each pass type identifier you create and these certificates expire every 398 days.

> ### Prepare a Certificate Signing Request

Reviewer: ```suggestion
**Prepare a Certificate Signing Request**
```
